### PR TITLE
New Workflow - Enrich indicators with Recorded Future from IDR Alert

### PR DESCRIPTION
**Pull request by dsliwinski-r7**
New Workflow - Enrich indicators with Recorded Future from IDR Alert

## Requirements

Workflow contributors, verify you have completed the following items by checking them off:

### Checklist
- [ ] Workflow verified as working in a test
- [ ] Workflow bundle includes `Name_of_Workflow.icon` file, `help.md` file, `workflow.spec.yaml` file, and `screenshots` directory
- [ ] Workflow title should follow general [Title of Articles](http://grammar.yourdictionary.com/capitalization/rules-for-capitalization-in-titles.html) format
- [ ] Workflow includes a detailed Description
- [ ] Workflow step names follow [Title of Articles](http://grammar.yourdictionary.com/capitalization/rules-for-capitalization-in-titles.html) format
- [ ] Triggers, Decisions, and other workflow steps include descriptions where needed
- [ ] Workflow includes at least one `Final Report` artifact that summarizes workflow execution details
- [ ] Workflow bundle (directory) is named after the title of the workflow in snake case with first letters capitalized, e.g. `workflows/Name_of_Workflow/`
- [ ] `help.md` file includes Description, Key Features, Requirements, Setup, Technical Details, Version History, and References
- [ ] `workflow.spec.yaml` file includes `products`, `name`, `title`, `description`, `use_cases` tags, `keywords` tags, `source_url`, and the `name` for each screenshot matches a PNG image in the `screenshots` directory

### Assessment

<details>

```
[*] Validating workflow at .

[*] Running Integration Validators...
[*] Executing validator WorkflowDirectoryNameMatchValidator
[*] Executing validator WorkflowFilesValidator
[*] Executing validator WorkflowHelpValidator
[*] Executing validator WorkflowChangelogValidator
[*] Executing validator WorkflowVendorValidator
[*] Executing validator WorkflowVersionValidator
[*] Executing validator WorkflowExtensionValidator
[*] Executing validator WorkflowSupportValidator
[*] Executing validator WorkflowPNGHashValidator
[*] Executing validator WorkflowICONFileNameValidator
[*] Executing validator WorkflowScreenshotValidator
[*] Executing validator WorkflowTitleValidator
[*] Executing validator WorkflowDescriptionValidator
[*] Executing validator WorkflowNameValidator
[*] Executing validator WorkflowProfanityValidator
[*] Executing validator WorkflowHelpPluginUtilizationValidator
[*] Executing validator WorkflowICONFileValidator
[*] Workflow successfully validated!

----
[*] Total time elapsed: 93.221ms
```

<summary>
icon validate
</summary>
</details>

### Screenshots

Screenshots help your reviewers provide valuable feedback on your contribution. To add a screenshot here, simply drag and drop the PNG files from a file explorer.

#### Workflow Builder

workflow1
workflow2
workflow3
workflow4
workflow5
workflow6

#### Job

job1
job2

#### Artifact

idr_alert_summary
results1
results2
results3

### CODE REVIEW

File: workflows/Enrich_Indicators_with_Recorded_Future_from_IDR_Alert/help.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+This workflow enriches an IDR alert by performing a lookup on all domains, hashes, URLs and IPs in the investigation with Recorded Future.
+
+# Key Features
+
+* Perform a domain, hash, URL and IP lookup automatically from any IDR alert
+
+# Requirements
+
+* InsightConnect License
```
Comment (jschipp-r7) [inline workflows/Enrich_Indicators_with_Recorded_Future_from_IDR_Alert/help.md:11]:
```suggestion
```

File: workflows/Enrich_Indicators_with_Recorded_Future_from_IDR_Alert/help.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+* Recorded Future API Key and Account Credentials
+
+# Documentation
+
+## Setup
+
+* Download the workflow or clone the repository `git clone https://github.com/rapid7/insightconnet-workflows.git`
+* Login to InsightConnect, and “Import” the .icon file into the workflow builder
```
Comment (jschipp-r7) [inline workflows/Enrich_Indicators_with_Recorded_Future_from_IDR_Alert/help.md:19]: These first two lines can be replaced with

```suggestion
Import the workflow from the Rapid7 Extension Library and proceed through the Import Workflow wizard in InsightConnect. Import plugins, create or select connections, and rename the workflow as a part of the Import Workflow wizard as necessary.
```

**Review by jschipp-r7 — CHANGES_REQUESTED** [review]
We can remove the following two screenshots:

* workflows/Enrich_Indicators_with_Recorded_Future_from_IDR_Alert/screenshots/job.png
* workflows/Enrich_Indicators_with_Recorded_Future_from_IDR_Alert/screenshots/artifacts.png

And replace them with screenshots showing a successful workflow run in Slack with some of the output

**Comment by dsliwinski-r7** [comment]
> We can remove the following two screenshots:
> 
>     * workflows/Enrich_Indicators_with_Recorded_Future_from_IDR_Alert/screenshots/job.png
> 
>     * workflows/Enrich_Indicators_with_Recorded_Future_from_IDR_Alert/screenshots/artifacts.png
> 
> 
> And replace them with screenshots showing a successful workflow run in Slack with some of the output

Hi @jschipp-r7! This workflow is triggered by an IDR alert, so I can't run it in Slack. Do you want me to add a few steps in which the results will be posted on Slack (using Slack ChatOps) or do you want something else?

**Comment by jschipp-r7** [comment]
@dsliwinski-r7 ah, yes - show the successful output of the workflow then

**Comment by dsliwinski-r7** [comment]
@jschipp-r7 I've added screenshots showing the successful output of the workflow.

**Review by jschipp-r7 — CHANGES_REQUESTED** [review]
Can we improve the readability of the artifact card? and use bold for the keys


*Risk Summary*: 4 of 47 Risk Rules currently observed
*Risk Score*: 2


 and also format the larger object Evidence Details. You can do that with input templating.
https://docs.rapid7.com/insightconnect/format-strings-with-templates#template-print-all-json-object-items

```
{{#each [StepName].[JSON object variable]}}
{{#each .}}
  {{@key}} = {{.}}
{{/each}}
{{/each}}
```

Need to address this one
* https://github.com/rapid7/insightconnect-workflows/pull/385/files#r534531751

**Comment by dsliwinski-r7** [comment]
@jschipp-r7 Sorry about that. I was sure I had already solved it :)

I improved the readability of the artifacts and updated the screenshots.

**Review by jschipp-r7 — APPROVED** [review]

**Pull request merged**